### PR TITLE
Provide user profile in select account and consent screen

### PR DESCRIPTION
**Pull request by IniZio**
ref #4136 / DEV-1225

### CODE REVIEW

File: resources/authgear/templates/en/web/__user_profile_display_name.html [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{{- define "__user_profile_display_name.html" -}}
+{{- if .UserProfile.email -}}
+  {{- .UserProfile.email -}}
+{{- else if .UserProfile.phone_number -}}
+  {{- .UserProfile.phone_number -}}
+{{- else if .UserProfile.preferred_username -}}
+  {{- .UserProfile.preferred_username -}}
+{{- end -}}
+{{- end -}}
```
Comment (louischan-oursky) [inline resources/authgear/templates/en/web/__user_profile_display_name.html:9]: We do not need this file.

File: pkg/auth/handler/oauth/consent.go [hunk 2]
```diff
@@ -45,12 +47,18 @@ type ProtocolIdentityService interface {
...
+}
+
 type ConsentViewModel struct {
 	ClientName               string
 	ClientPolicyURI          string
 	ClientTOSURI             string
 	IsRequestingFullUserInfo bool
-	IdentityDisplayName      string
+	// Deprecated: Use UserProfile instead
+	IdentityDisplayName string
```
Comment (louischan-oursky) [inline pkg/auth/handler/oauth/consent.go:60]: Keep

File: resources/authgear/templates/en/web/authflowv2/select_account.html [hunk 3]
```diff
@@ -14,7 +14,7 @@ <h1 class="screen-title">
       {{ end }}
     </h1>
     <p class="screen-description">
-      {{ template "v2-select-account-description" (dict "IdentityDisplayName" $.IdentityDisplayName) }}
+      {{ template "v2-select-account-description" (dict "UserProfileDisplayName" (include "__user_profile_display_name.html" .)) }}
```
Comment (louischan-oursky) [inline resources/authgear/templates/en/web/authflowv2/select_account.html:17]:
```
(merge (dict) .UserProfile (dict "IdentityDisplayName" $.IdentityDisplayName))
```

Define UserProfile as

```
	Email               = "email"
	EmailVerified       = "email_verified"
	PhoneNumber         = "phone_number"
	PhoneNumberVerified = "phone_number_verified"
	PreferredUsername   = "preferred_username"
	FamilyName          = "family_name"
	GivenName           = "given_name"
	MiddleName          = "middle_name"
	Name                = "name"
	Nickname            = "nickname"
	Picture             = "picture"
	Profile             = "profile"
	Website             = "website"
	Gender              = "gender"
	Birthdate           = "birthdate"
	Zoneinfo            = "zoneinfo"
	Locale              = "locale"
```

- If the attribute is a string and the user does not have it, then output "nil" string
- If the attribute is a string and the user has it, then output the value.
- If the attribute is a boolean and the user does not have it, then output "nil" string
- If the attribute is a boolean and the user has it, then output the value.

Comment (IniZio) [reply resources/authgear/templates/en/web/authflowv2/select_account.html]: As per discussion, updated to add `xxx_is_present` flag to know a field's presence

**Review by louischan-oursky — APPROVED** [review]

**Pull request merged**